jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -939,7 +939,11 @@ public boolean checkContext(final String target, final Request baseRequest, fina
             return false;
 
         if (!checkContextPath(target))
+        {
+            baseRequest.setHandled(true);
```

Comment:
Unfortunately it can't be done this way, as context handlers can be chained and fall through to each other.

I think this is probably fixed in 9.3 as we have changed the way we log requests.